packages/devtools_app/lib/src/shared/table/table.dart

```diff
@@ -811,6 +806,31 @@ class _TableState<T> extends State<_Table<T>> with AutoDisposeMixin {
     scrollController = widget.tableController.verticalScrollController!;
 
     pinnedScrollController = ScrollController();
+  }
+
+  @override
+  void didUpdateWidget(covariant _Table<T> oldWidget) {
+    super.didUpdateWidget(oldWidget);
+
+    final dataChanged = widget.tableController.tableData !=
+            oldWidget.tableController.tableData ||
+        widget.selectionNotifier != oldWidget.selectionNotifier ||
+        widget.activeSearchMatchNotifier != oldWidget.activeSearchMatchNotifier;
+
+    if (dataChanged) {
+      cancelListeners();
+      _initData();
+    }
+  }
+
+  void _initData() {
```

Comment:
nit: rename to _init, _initHelper, _initDataAndAddListeners since this is doing more than just initializing the data.